gax.go

```diff
@@ -1,191 +1,169 @@
-	Len() int
-
-	// GetData returns the i-th element in the current response data.
-	GetData(i int) interface{}
-
-	// NextPage updates the page token of the current request from the response data. This should
-	// return io.EOF error if it doesn't have the page token anymore, and return other non-nil
-	// errors if something goes wrong.
-	NextPage() error
-}
-
-// PageStream iterates over the elements in the page-streaming data. |iter| is invoked for each
-// of the elements, and the iteration will stop when it returns false.
-func PageStream(ctx context.Context, streamable PageStreamable, iter func(interface{}) bool, opts ...CallOption) error {
-	for {
-		// TODO(mukai): tweak callOpt for iteration?
-		err := streamable.ApiCall(ctx, opts...)
-		// io.EOF might happen when the initial ApiCall reaches to the end for streaming gRPC calls.
-		// This means the requested resource is simply empty, therefore return nil instead of errors.
-		if err == io.EOF {
-			return nil
-		} else if err != nil {
-			return err
-		}
-		length := streamable.Len()
-		for i := 0; i < length; i++ {
-			if !iter(streamable.GetData(i)) {
-				return nil
-			}
-		}
-		if err := streamable.NextPage(); err == io.EOF {
-			return nil
-		} else if err != nil {
-			return err
-		}
+func (opts callOptions) Resolve(s *CallSettings) *CallSettings {
+	for _, opt := range opts {
+		opt.Resolve(s)
 	}
+	return s
 }
 
-// Head returns the first element from the page stream.
-func Head(ctx context.Context, streamable PageStreamable, opts ...CallOption) (result interface{}, err error) {
-	called := false
-	err = PageStream(ctx, streamable, func(element interface{}) bool {
-		called = true
-		result = element
-		return false
-	}, opts...)
-	if !called {
-		return nil, errors.New("PageStreamable is empty.")
-	}
-	return result, err
+// Encapsulates the call settings for a particular API call.
+type CallSettings struct {
+	// The client-side timeout for API calls.
+	Timeout time.Duration
+	// Determines whether and how to retry on transient errors.
+	RetrySettings RetrySettings
 }
 
-// Take returns the first |count| elements in the page stream.
-func Take(ctx context.Context, streamable PageStreamable, count int, opts ...CallOption) ([]interface{}, error) {
-	if count <= 0 {
-		return nil, nil
-	}
-	result := make([]interface{}, 0, count)
-	i := 0
-	err := PageStream(ctx, streamable, func(element interface{}) bool {
-		result = append(result, element)
-		i++
-		return i < count
-	}, opts...)
-	return result, err
-}
-
-// ToArray converts the page stream into a single array.
-func ToArray(ctx context.Context, streamable PageStreamable, opts ...CallOption) ([]interface{}, error) {
-	var result []interface{}
-	err := PageStream(ctx, streamable, func(element interface{}) bool {
-		result = append(result, element)
-		return true
-	}, opts...)
-	return result, err
+func (s *CallSettings) Resolve(c *CallSettings) {
+	c.Timeout = s.Timeout
+	c.RetrySettings = s.RetrySettings
```

Comment:
Obsolete, but good call - I originally had that as a pointer, and didn't notice after I changed it.